kola/tests/update/update.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package update
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-omaha/omaha"
+
+	"github.com/coreos/mantle/kola"
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/coreos/mantle/platform/local"
+	"github.com/coreos/mantle/platform/machine/qemu"
+	"github.com/coreos/mantle/sdk"
+	"github.com/coreos/mantle/util"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Name:        "coreos.update.updatepayload",
+		Run:         updatepayload,
+		ClusterSize: 1,
+		NativeFuncs: map[string]func() error{
+			"Omaha": Serve,
+		},
+	})
+}
+
+func Serve() error {
+	go func() {
+		omahaserver, err := omaha.NewTrivialServer(":34567")
+		if err != nil {
+			fmt.Printf("creating trivial omaha server: %v\n", err)
+		}
+
+		omahawrapper := local.OmahaWrapper{TrivialServer: omahaserver}
+
+		if err = omahawrapper.AddPackage("/updates/update.gz", "update.gz"); err != nil {
+			fmt.Printf("bad payload: %v", err)
+		}
+
+		go omahawrapper.Serve()
+	}()
+
+	select {}
+}
+
+func updatepayload(c cluster.TestCluster) {
+	if kola.UpdatePayload == "" {
+		c.Skip("no update payload provided")
+	}
+
+	// create the actual test machine, the machine
+	// that is created by the test registration is
+	// used to host the omaha server if running on
+	// a non-qemu platform
+	m, err := c.NewMachine(conf.Ignition(`{"ignition":{"version": "2.1.0"}}`))
```

Comment:
`c.NewMachine(nil)` is sufficient (though that will currently pass a 2.0.0 config, if that matters).